File: aoc-2015/src/test/java/net/bbelovic/adventofcode/day5/Day5PuzzlePart2Test.java
package net.bbelovic.adventofcode.day5;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day5PuzzlePart2Test {

    @ParameterizedTest
    @MethodSource("testData")
    void
    should_tell_whether_input_string_is_nice(String input, boolean expectedResult) {
        Day5PuzzlePart2 day5 = new Day5PuzzlePart2();
        boolean actualResult = day5.solve(input);
        Assertions.assertEquals(expectedResult, actualResult);
    }

    private static Stream<Arguments> testData() {
        return Stream.of(
                Arguments.of("xxyxx", true),
                Arguments.of("qjhvhtzxzqqjkmpb", true),
                Arguments.of("uurcxstgmygtbstg", false),
                Arguments.of("ieodomkazucvgmuy", false),
                Arguments.of("urrvucyrzzzooxhx", false)
        );
    }
}

File: aoc-2020/src/test/kotlin/net/bbelovic/adventofcode/year2020/day5/Day5PuzzlePart2Test.kt
package net.bbelovic.adventofcode.year2020.day5

import net.bbelovic.adventofcode.InputReader
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day5PuzzlePart2Test {
    @Test
    fun `should find seat ID`() {
        val lines = InputReader().readAllLines("input5.txt")
        val actualSeatId = Day5PuzzlePart2().solve(lines)
        assertThat(actualSeatId).isEqualTo(615L)
    }
}